clean: Refactor details on how coding standards work DOCS-556 (#1739)
codacy/docs: docs/organizations/using-coding-standards.md

# Using coding standards

Coding standards help you ensure that Codacy analyzes multiple repositories with the same tool and code pattern configurations. For example, you can use a coding standard to ensure that a group of repositories follow the same security rules or coding conventions.

Create coding standards on your organization to define different tool and code pattern configurations that apply to a set of programming languages. After creating a coding standard, apply it to your repositories:

-   Applying a coding standard to a repository only affects the configurations of the tools that are included in the coding standard, while the remaining tool and code pattern configurations of the repository remain unchanged.
-   If you later edit the tool and code pattern configurations in a coding standard, the repositories following that coding standard automatically start using the updated configurations on the next analysis.
-   When you customize the tools or code patterns of a repository that follows a coding standard, Codacy warns you that the repository will stop following the coding standard and asks for your confirmation.

> -   Each repository can only follow one coding standard at a time. Applying a new coding standard to a repository <span class="skip-vale">unassigns</span> any previously applied coding standard.

Reviewer: @prcr I really like the way you refactored this introduction. 👍 You managed to gather several independent facts into a common umbrella, which is "applying coding standards to repositories". 💡 🤓 

My only suggestion would be to move this last bullet to the top, as it seems to me more general and easy-to-understand information about applying a coding standard to repositories. The other three are more detailed, they could come next in the same order.

Author: Indeed, now that I re-read the list this makes perfect sense! I already updated the order. :+1:

> Optionally, set a default coding standard to apply it automatically to all new repositories that you add to Codacy. If no coding standard is set as default, new repositories won't follow any coding standard and will use the Codacy default configurations instead.

Reviewer 2: Could this work in the present tense?

```suggestion
Optionally, set a default coding standard to apply it automatically to all new repositories that you add to Codacy. If no coding standard is set as default, new repositories don't follow any coding standard and use the Codacy default configurations instead.
```

1.  Click the button **Save and apply standard** on the repository selection page to save your changes to the coding standard.

    Codacy will start using the updated coding standard on the next analysis of each selected repository.

>     !!! note
>         If you stop applying a coding standard to any repositories, Codacy restores the previous code pattern configurations for those repositories. However, the repositories keep the toggle status of the tools from the coding standard.

Reviewer 2: There are some tough concepts to convey here!

* "Toggle status" is a bit unclear; I was unsure what it referred to. How about "activation status"?
* I'd use the singular. Personal preference perhaps, but it feels more readable to me.
* I'd also reword to have one subject, (Codacy does X, but also Y) instead of two (Codacy does X, the repositories do Y).

```suggestion
    !!! note
        If you stop applying a coding standard to a repository, Codacy restores the previous code pattern configurations of that repository, but keeps the activation status of the tools as defined by the coding standard.
```

Author: These are nice edits for simplifying the text! :star_struck:

1.  Click the trash can icon on the coding standard card and confirm.

    !!! note

>         If you delete a coding standard, Codacy restores the previous code pattern configurations for any repositories following the coding standard. However, the repositories keep the toggle status of the tools from the coding standard.

Reviewer 2: See [comment above](https://github.com/codacy/docs/pull/1739/files#r1206757043).

Author: I'm editing this note like this:

```suggestion
    !!! note
        If you delete a coding standard Codacy restores the previous code pattern configurations for any repositories following the coding standard, but keeps the tool activation status defined by the coding standard.
```

Reviewer 2: Looking good!
